packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -174,158 +190,76 @@ export class EmrContainersStartJobRun extends sfn.TaskStateBase {
     };
   }
 
-
   /**
    * Render the EMR Containers ConfigurationProperty as JSON
    *
    * @param property
    */
-  private ApplicationConfigPropertyToJson(property: ApplicationConfiguration) {
+  private applicationConfigPropertyToJson(property: ApplicationConfiguration) {
     return {
       Classification: cdk.stringToCloudFormation(property.classification.classificationStatement),
-      Properties: cdk.objectToCloudFormation(property.properties),
-      Configurations: cdk.listMapper(this.ApplicationConfigPropertyToJson)(property.nestedConfig),
+      Properties: property.properties ? cdk.objectToCloudFormation(property.properties) : undefined,
+      Configurations: property.nestedConfig ? cdk.listMapper(this.applicationConfigPropertyToJson)(property.nestedConfig) : undefined,
     };
   }
 
-  private renderTags(tags: { [key: string]: any } | undefined): { [key: string]: any } {
-    return tags ? { Tags: Object.keys(tags).map((key) => ({ Key: key, Value: tags[key] })) } : {};
+  private isArrayOfStrings(value: any): boolean {
+    return Array.isArray(value) && value.every(item => typeof item === 'string');
   }
 
-  /**
-   * Creates a Cloudwatch log group
-   *
-   * @returns name if user enables props.monitoring.logging to be true
-   **/
-  private renderLogGroupName(): string {
-    return new logs.LogGroup(this, 'Emr Containers Default Cloudwatch Log Group').logGroupName;
-  }
-
-  /**
-   * Creates a S3 bucket
-   *
-   * @returns an automatically generated bucket name
-   **/
-  private renderS3BucketName(): string {
-    return new s3.Bucket(this, 'Emr Containers Default S3 Bucket').bucketName;
+  private renderTags(tags?: { [key: string]: any }): { [key: string]: any } {
+    return tags ? { Tags: Object.keys(tags).map((key) => ({ Key: key, Value: tags[key] })) } : {};
   }
 
   // https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/creating-job-execution-role.html
   private createJobExecutionRole(): iam.Role {
-    let jobExecutionRole: iam.Role = new iam.Role(this, 'Job-Execution-Role', {
-      assumedBy: new iam.ServicePrincipal('emr-containers.amazonaws.com'),
+    const jobExecutionRole = new iam.Role(this, 'Job-Execution-Role', {
+      assumedBy: new iam.CompositePrincipal(
+        new iam.ServicePrincipal('emr-containers.amazonaws.com'),
+        new iam.ServicePrincipal('states.amazonaws.com'),
+      ),
     });
 
-    // allow job-execution-role to be used by Step Functions
-    jobExecutionRole.assumeRolePolicy?.addStatements(
+    jobExecutionRole.addToPrincipalPolicy(
```

Comment:
Users would already have these statements in their job execution role if they do decide to pass in their own role. What about the case when these policy statements already exist within the role?

Reply:
There is no reason to believe that the permissions are already granted to the job execution role. In fact, the user should assume that we will grant all necessary permissions. There is no issue with granting the permissions multiple times, it will essentially be a no-op from IAM's perspective